### PR TITLE
Delete password reset verification key after it is used

### LINKED ISSUE #2512: Password reset codes should only be usable once
In #2509, it was noticed that the password reset link continues to work in perpetuity, even after being used.

The ability to password reset using that link should expire after the link is used.

### PR DESCRIPTION
**Pull request by cgolubi1**
* Fixes #2512

### CODE REVIEW

**Comment by cgolubi1** [comment]
I tested this by hand, and added a unit test.  I think it's pretty straightforward and you probably don't need a dev site, but let me know if you want one anyway.

File: src/engine/BMInterfaceNewuser.php [hunk 1]
```diff
@@ -299,6 +298,15 @@ public function reset_password($playerId, $playerKey, $password) {
             $statement->execute(array(':id' => $playerId,
                                       ':password' => $passwordHash));
 
+            // Now delete the verification key so it can't be reused
+            $query =
+                'DELETE FROM player_reset_verification ' .
+                'WHERE player_id = :player_id ' .
```
Comment (blackshadowshade) [inline src/engine/BMInterfaceNewuser.php]: What about clearing ALL verification keys for this player when successfully using one?

**Comment by blackshadowshade** [comment]
I'm happy to accept this pull request after my comment is addressed (either in the positive or the negative).

File: src/engine/BMInterfaceNewuser.php (hunk 1 above)
Comment (cgolubi1) [reply src/engine/BMInterfaceNewuser.php]: Okay.

**Comment by blackshadowshade** [comment]
Looks good. I'll accept this when CircleCI is done.

**Pull request merged**